### ISSUE TITLE
IDE0071 invalid simplification suggested with ReadOnlySpan that generates CS0029

### ISSUE THREAD

**Issue by eduherminio** [dotnet/roslyn#44112]
**Version Used**: 

16.5.0

**Steps to Reproduce**:

```c#
var span = "44112".AsSpan();

var uri = new Uri($"https://github.com/dotnet/roslyn/issues/{span.ToString()}");
```

**Expected Behavior**:

No simplification suggested.

**Actual Behavior**:

Invalid simplification suggested, which generates CS0029
`Cannot implicitly convert type 'System.ReadOnlySpan<char>' to 'object'`.

**Comment by sharwell** [comment]
Duplicate of #42936

**Comment by sharwell** [comment]
@eduherminio This should be fixed starting in 16.6 Preview 3 👍
